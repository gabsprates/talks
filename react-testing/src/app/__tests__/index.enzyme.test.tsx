import React from "react";
import { mount } from "enzyme";
import { App } from "..";
import styleDevDetails from "../../devDetails/index.scss";

jest.mock("../../services/github");

describe("Component: App", () => {
  it("should render without errors", () => {
    const wrapper = mount(<App />);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should open dev details and clean after click the button", () => {
    const wrapper = mount(<App />);

    let button = wrapper.find("button");
    expect(button.exists()).toBeFalsy();

    let user = wrapper.find("li").first();
    user.simulate("click");

    let devDetails = wrapper.find(`.${styleDevDetails.devDetails}`);
    expect(devDetails.exists()).toBeTruthy();

    button = wrapper.find("button");
    button.simulate("click");

    devDetails = wrapper.find(`.${styleDevDetails.devDetails}`);
    expect(devDetails.exists()).toBeFalsy();
    expect(wrapper.find("button").exists()).toBeFalsy();
  });
});
