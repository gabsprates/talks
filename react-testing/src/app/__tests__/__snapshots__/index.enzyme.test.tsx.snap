// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: App should render without errors 1`] = `"<section class=\\"app\\"><h1>Zone Of Front-Enders</h1><div class=\\"columns\\"><aside><ul><li role=\\"item-user\\"><div class=\\"user\\"><strong class=\\"name\\">Andr√© Simoes</strong><div class=\\"github\\">Andre-rsfc</div></div></li><li role=\\"item-user\\"><div class=\\"user\\"><strong class=\\"name\\">Artur Bani</strong><div class=\\"github\\"></div></div></li><li role=\\"item-user\\"><div class=\\"user\\"><strong class=\\"name\\">Carolina Sampaio</strong><div class=\\"github\\">carolinasamp</div></div></li><li role=\\"item-user\\"><div class=\\"user\\"><strong class=\\"name\\">Clarisse Silva</strong><div class=\\"github\\"></div></div></li><li role=\\"item-user\\"><div class=\\"user\\"><strong class=\\"name\\">Gabriel Ataide</strong><div class=\\"github\\">gabecar</div></div></li><li role=\\"item-user\\"><div class=\\"user\\"><strong class=\\"name\\">Gabriel de Paula</strong><div class=\\"github\\">gabriel949</div></div></li><li role=\\"item-user\\"><div class=\\"user\\"><strong class=\\"name\\">Gabriel Prates</strong><div class=\\"github\\">gabsprates</div></div></li><li role=\\"item-user\\"><div class=\\"user\\"><strong class=\\"name\\">Ian Rodrigues</strong><div class=\\"github\\">IanPaixao</div></div></li><li role=\\"item-user\\"><div class=\\"user\\"><strong class=\\"name\\">Izabela Bittencourt</strong><div class=\\"github\\">belabittencourt</div></div></li><li role=\\"item-user\\"><div class=\\"user\\"><strong class=\\"name\\">Pedro Barbosa</strong><div class=\\"github\\"></div></div></li><li role=\\"item-user\\"><div class=\\"user\\"><strong class=\\"name\\">Saulo Vinicius</strong><div class=\\"github\\">sauloviniciusf</div></div></li></ul></aside></div></section>"`;
